### PR TITLE
chore: use osctl cluster --wait in basic-integration

### PR DESCRIPTION
**Pull request by smira**
Added flag to workaround CI/CD issue to use direct connection to docker
container (init node) instead of 127.0.0.1 passthrough (default is still
using 127.0.0.1).

Signed-off-by: Andrey Smirnov <smirnov.andrey@gmail.com>

### CODE REVIEW

File: pkg/config/types/v1alpha1/generate/controlplane.go [hunk 1]
```diff
@@ -17,7 +17,7 @@ func controlPlaneUd(in *Input) (string, error) {
 		MachineType:     "controlplane",
 		MachineToken:    in.TrustdInfo.Token,
 		MachineCA:       in.Certs.OS,
-		MachineCertSANs: []string{},
+		MachineCertSANs: in.AdditionalSubjectAltNames,
```
Comment (smira) [inline pkg/config/types/v1alpha1/generate/controlplane.go]: I'm not sure if `AdditionalSubjectAltNames` can be applied here for `apid` cert? Or we need a different setting in `generate.Input`?

Comment (andrewrynhard) [reply pkg/config/types/v1alpha1/generate/controlplane.go]: We have two sets of SANs in our config... we have one for k8s and one for Talos. I don't think we have a way to set the Talos SANs in the `Input` though.

Comment (smira) [reply pkg/config/types/v1alpha1/generate/controlplane.go]: So should I introduce separate field for Talos SANs or fine to re-use?

Comment (andrewrynhard) [reply pkg/config/types/v1alpha1/generate/controlplane.go]: I think we will have to introduce another flag.

Comment (andrewrynhard) [reply pkg/config/types/v1alpha1/generate/controlplane.go]: Can't think of a better way at the moment... maybe once @rsmitty implements the ability to specify files for the user data then maybe we could generate the configs, modify them, and then create the cluster.

File: hack/test/basic-integration.sh [hunk 2]
```diff
@@ -3,93 +3,32 @@
...
-export KUBECONFIG="${TMP}/kubeconfig"
-export TIMEOUT=300
-export OSCTL="${PWD}/${ARTIFACTS}/osctl-linux-amd64"
-export INTEGRATIONTEST="${PWD}/bin/integration-test"
+TALOS_IMG="docker.io/autonomy/talos:${TAG}"
+OSCTL="${PWD}/${ARTIFACTS}/osctl-linux-amd64"
+INTEGRATIONTEST="${PWD}/bin/integration-test"
+
+
+case "${CI:-false}" in
```
Comment (andrewrynhard) [inline hack/test/basic-integration.sh]: I don't see where CI is getting set. Does Drone set it?

Comment (smira) [reply hack/test/basic-integration.sh]: yep, it's one of the Drone environment variables

File: pkg/config/types/v1alpha1/generate/controlplane.go (hunk 1 above)
Comment (smira) [reply pkg/config/types/v1alpha1/generate/controlplane.go]: Added `AdditionalMachineCertSANs` :)

**Pull request merged**